org.eclipse.jdt.debug.ui/ui/org/eclipse/jdt/internal/debug/ui/jres/JREsPreferencePage.java

```diff
@@ -87,6 +92,8 @@ public JREsPreferencePage() {
 	 */
 	@Override
 	public void init(IWorkbench workbench) {
+		setPreferenceStore(new ScopedPreferenceStore(InstanceScope.INSTANCE,
```

Comment:
Haven't debugged that, but why do we need to set a preference store for the **existing** preference page? How did it worked before?

Reply:
The preference page uses other JDT APIs to interact with the configuration, see performOK. Setting the store allows `super.performOK` to store this new preference as usual.